naga/src/back/glsl/mod.rs

```diff
@@ -3756,11 +3806,37 @@ impl<'a, W: Write> Writer<'a, W> {
                     Mf::FirstTrailingBit => "findLSB",
                     Mf::FirstLeadingBit => "findMSB",
                     // data packing
-                    Mf::Pack4x8snorm => "packSnorm4x8",
-                    Mf::Pack4x8unorm => "packUnorm4x8",
-                    Mf::Pack2x16snorm => "packSnorm2x16",
-                    Mf::Pack2x16unorm => "packUnorm2x16",
-                    Mf::Pack2x16float => "packHalf2x16",
+                    Mf::Pack4x8snorm if self.options.version.supports_pack_unpack_4x8() => {
```

Comment:
Please match the variants once and move the `if`s inside each block, this is the existing style and I find it easier to read.